### PR TITLE
Set dart SDK version

### PR DESCRIPTION
**Pull request by solid-oleksandrhordiienko**

### CODE REVIEW

**Comment by solid-oleksandrhordiienko** [comment]
Before merging need to change if condition 
From:
`github.repository	== 'solid-software/flutter_project_template'`
To:
`github.repository	!= 'solid-software/flutter_project_template'`

**Comment by illia-romanenko** [comment]
Let's try to fix this line based replace with something more intelligent. @solid-yuriiprykhodko should be able to help.

File: .github/workflows/generate-template.yaml [hunk 1]
```diff
@@ -65,6 +63,10 @@ jobs:
           flutter pub add --dev solid_lints
           flutter pub get
 
+      - name: Set dart sdk version
+        run: |
+          a=`sed '22!d' pubspec.yaml` && sed -i "s/$a/  sdk: '>=2.17.0 <4.0.0'/1" pubspec.yaml
```
Comment (solid-yuriiprykhodko) [inline .github/workflows/generate-template.yaml]: Ah, the good ol' bash magic.
You write it and it works, but then it breaks and nobody knows what it even does.

@solid-oleksandrhordiienko Could you please give some context on what we're trying to accomplish here?

Comment (solid-yuriiprykhodko) [reply .github/workflows/generate-template.yaml]: I see -- by default the template generates `sdk: '>=2.19.0 <3.0.0'` and we're trying to widen the constraints to be `'>=2.17.0 <4.0.0'`. But __why__ do we need this?

**Comment by solid-yuriiprykhodko** [comment]
Main issue is with Dart Code Metrics action:

- https://github.com/dart-code-checker/dart-code-metrics-action/issues/87
- https://github.com/dart-code-checker/dart-code-metrics-action/issues/161

I can suggest the following solutions:

- Use v3 of dart_code_metrics
- Use dart_code_metrics CLI directly.

**Pull request closed without merging**